ddtrace/tracer/tracer.go

```diff
@@ -57,6 +60,36 @@ type tracer struct {
 	pid string
 }
 
+type statsdClient interface {
```

Comment:
Sorry for the back and forth here, but this should be in `metrics.go`. We should consolidate all statsd related code together. That seems like the right place.